Sources/Yams/Constructor.swift

```diff
@@ -185,50 +187,94 @@ extension Double: ScalarConstructible {
     }
 }
 
+extension Float: ScalarConstructible {
+    public static func construct(from node: Node) -> Float? {
+        assert(node.isScalar) // swiftlint:disable:next force_unwrapping
+        var scalar = node.scalar!.string
+        switch scalar {
+        case ".inf", ".Inf", ".INF", "+.inf", "+.Inf", "+.INF":
+            return Float.infinity
```

Comment:
Can omit the type here and below: (`return .infinity`, `return -.infinity`, `return .nan`). Same with the `Double` implementation.